hydromt_fiat/workflows/hazard.py

```diff
@@ -308,71 +327,97 @@ def fstrip(x):
             #             "var": None if not hasattr(self, "var_lst") else self.var_lst[idx],
             #             "chunks": "auto" if chunks == "auto" else self.chunks_lst[idx],
             #            }
-
-            da = da.expand_dims({"rp": [da_rp]}, axis=0)
+            
+            
 
             model_fiat.set_maps(da, da_name)
             # post = f"(rp {da_rp})" if rp is not None and risk_output else ""
             post = f"(rp {da_rp})" if risk_output else ""
             model_fiat.logger.info(f"Added {hazard_type} hazard map: {da_name} {post}")
 
-            # new_da = xr.concat([new_da, da], dim='rp')
+            # new_da = xr.concat([new_da, da], dim='rp')   
 
             # import numpy as np
             # new_da = xr.DataArray(np.zeros_like(da), dims=da.dims, coords=da.coords).rename('new_dataframe')
 
-        # # model_fiat.maps.to_netcdf("test_hazard_1/test.nc")
 
-        # #model_fiat.set_maps(catalog, "all")
-        maps = model_fiat.maps
-        list_keys = list(maps.keys())
-        maps_0 = maps[list_keys[0]].rename("risk")
-        list_keys.pop(0)
-
-        for idx, x in enumerate(list_keys):
-            key_name = list_keys[idx]
-            layer = maps[key_name]
-            maps_0 = xr.concat([maps_0, layer], dim="rp")
-
-        # new_da = maps_0.to_dataset(name='RISK')
-        # new_da.attrs = {  "returnperiod": list(list_rp),
-        #                   "type":self.map_type_lst,
-        #                   'name':list_names}
-        
-        if not risk_output:
-            # new_da = maps_0.drop_dims('rp')
-            new_da = maps_0.to_dataset(name="EVENT")
-            new_da.attrs = {
-                "returnperiod": list(list_rp),
-                "type": self.map_type_lst,
-                "name": list_names,
-                "Analysis": "Event base",
-            }
-
-        else:
+        if risk_output:
+            maps = model_fiat.maps
+            list_keys = list(maps.keys())
+            maps_0 = maps[list_keys[0]].rename('risk')
+            list_keys.pop(0)
+
+            for idx, x in enumerate(list_keys):
+                key_name = list_keys[idx]
+                layer = maps[key_name]
+                maps_0 = xr.concat([maps_0, layer], dim='rp') 
+
             new_da = maps_0.to_dataset(name='RISK')
             new_da.attrs = {  "returnperiod": list(list_rp),
                             "type":self.map_type_lst,
                             'name':list_names,
                             "Analysis": "Risk"}  
 
-        model_fiat.hazard = new_da
-        model_fiat.set_maps(model_fiat.hazard, 'HydroMT_Fiat_hazard')
+            model_fiat.hazard = new_da
+            model_fiat.set_maps(model_fiat.hazard, 'HydroMT_Fiat_hazard')
```

Comment:
Where and how is this HydroMT_Fiat_hazard name used?

Reply:
This part is also not needed, I made this mainly to have an attribute .hazard in the model that would trigger the writing component (if model.hazard then write in the toml file)

The line can be simplified as: model_fiat.set_maps(new_da, 'key of the map'). The key of the map can also be "risk_maps" we can define it when having a look into the outputs of the model builder.